gax/src/main/java/com/google/api/gax/retrying/NonCancellableFuture.java

```diff
@@ -51,4 +51,14 @@ public boolean cancel(boolean mayInterruptIfRunning) {
   void cancelPrivately() {
     super.cancel(false);
   }
+
+  @Override
```

Comment:
Can you please change this class to the following (and then replace:

```java
@InternalApi
@BetaApi
public final class NonCancellableFuture<ResponseT> extends AbstractApiFuture<ResponseT> {
  @Override
  public boolean cancel(boolean mayInterruptIfRunning) {
    return false;
  }
  
  @Override
  public boolean set(ResponseT value) {
    return false;
  }

  @Override
  public boolean setException(Throwable throwable) {
    return false;
  }

  void cancelPrivately() {
    super.cancel(false);
  }

  boolean setPrivately(ResponseT value) {
    return super.set(value);
  }

  boolean setExceptionPrivately(Throwable throwable) {
    return super.setException(throwable);
  }
}

```

and then call the package-private methods in `BasicRetryingFuture.setAttemptResult()`